### PR TITLE
Migration to serialization (`:settings` related)

### LINKED ISSUE #499: Switch from Named to mkstemp
https://github.com/ansible/ansible-navigator/blob/b7d6fd223f9f55fb5d176b60a687782135a9ae4b/ansible_navigator/actions/open_file.py#L91

- credit @abadger 

So one issue is portability, there's a race where the filehandle opened by `tempfile.NamedTemporaryFile()` hasn't been closed yet when `open(filename, "w")` is called.  On systems where the file can't be opened twice at the same time, that will fail.

The other one is security.  Because of the same race, sometimes the filename will be deleted before `open(filename, "w")` is called.  That gives an attacker a brief window to put their own file, symlink, etc, in place.  IIRC, a symlink would allow the attacker to overwrite one of the user's file with the data, corrupting that file.

### PR DESCRIPTION
**Pull request by cidrblock**
Migrate to the serialization function(s) across the project.

- Added `serialize_write` and `serialize_write_temp` to central the logic into one place.
- Necessary changes to UI, open, run, and write actions
- Ensure everything coming and going from serialize is now a ``Path``
- Updated tests to use serialize rather than json directly for additional coverage
- Made an attempt to narrow down the types, but it tricky when we allow `:{{ some_nested_thing }}`
- Removed the double open of files
- Additional testing done locally accessing nested data, opening and writing files
- Switch to SafeDumper for yaml, add related tests for both serialization formats.

Closes #971 
Closes #499

### CODE REVIEW

File: src/ansible_navigator/actions/open_file.py [hunk 1]
```diff
@@ -85,7 +87,7 @@ def run(self, interaction: Interaction, app: AppPublic) -> None:
         if something:
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply src/ansible_navigator/actions/open_file.py:87]: yeah probably should have just called it `the_thing_the_user_wants_to_open` :)

File: src/ansible_navigator/actions/open_file.py [hunk 2]
```diff
@@ -85,7 +87,7 @@ def run(self, interaction: Interaction, app: AppPublic) -> None:
...
             parts = something.rsplit(":", 1)
             if os.path.isfile(parts[0]):
-                filename = parts[0]
+                filename = Path(parts[0])
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply src/ansible_navigator/actions/open_file.py:90]: Can't assign it to filename just yet, but we can call is "possible"

File: src/ansible_navigator/actions/run.py [hunk 3]
```diff
@@ -849,16 +852,21 @@ def write_artifact(self, filename: Optional[str] = None) -> None:
...
+                    "plays": self._plays.value,
+                    "stdout": self.stdout,
+                    "status": status,
+                    "status_color": status_color,
+                }
+                serialize_write_file(
+                    content=artifact,
+                    content_view=ContentView.NORMAL,
+                    file_mode="w",
+                    file=Path(filename),
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply src/ansible_navigator/actions/run.py:866]: yes. just trying to maintain some limit to how deep this went.

File: src/ansible_navigator/actions/write_file.py [hunk 4]
```diff
@@ -83,11 +86,20 @@ def run(self, interaction: Interaction, app: AppPublic) -> None:
...
+            )
         elif write_as == "json":
-            with open(os.path.abspath(filename), file_mode, encoding="utf-8") as fh:
-                json_dump(obj, fh)
-                fh.write("\n")
-
+            serialize_write_file(
+                content=obj,
+                content_view=ContentView.NORMAL,
+                file=Path(filename),
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply src/ansible_navigator/actions/write_file.py:100]: same

File: src/ansible_navigator/utils/serialize.py [hunk 5]
```diff
@@ -42,36 +45,111 @@ class SerializationFormat(Enum):
     from yaml import SafeLoader  # type: ignore[misc] # noqa: F401
 # pylint: enable=unused-import
 
+ContentType = Union["ContentBase", bool, Dict[str, Any], float, int, List[Any], str]
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply src/ansible_navigator/utils/serialize.py:48]: I've switched to SafeDumper for yaml, and added a try/except for both json/yaml and serialize.  

Additional tests for all these scenarios.

The error will be returned to the user, and they are asked to log an issue.

LMK what you think

(BTW,  I'm working on GH rule that will auto assign these to you :) )

**Review by ganeshrn — APPROVED** [review]

**Pull request merged**